src/material/form-field/_mdc-text-field-structure-overrides.scss

```diff
@@ -20,11 +20,14 @@
     border: none;
   }
 
-  // In order to ensure proper alignment of the floating label, we reset its line-height.
-  // The line-height is not important as the element is absolutely positioned and only has one line
-  // of text.
   .mat-mdc-form-field .mat-mdc-floating-label.mdc-floating-label {
+    // In order to ensure proper alignment of the floating label, we reset its line-height.
+    // The line-height is not important as the element is absolutely positioned and only has one
+    // line of text.
     line-height: normal;
+    // This makes migration from the legacy form-field easier for tests that were depending on
+    // clicking the label to focus the input.
+    pointer-events: all;
```

Comment:
You might also need to add `cursor: default` otherwise the label might have edit cursor when hovered.

Reply:
I think we actually want an edit cursor since the label overlaps with the input